src/test/java/org/apache/ibatis/cache/CacheKeyTest.java

```diff
@@ -80,4 +88,31 @@ public void shouldTestCacheKeysWithBinaryArrays() throws Exception {
     assertTrue(key1.equals(key2));
   }
 
+  @Test (expected = NotSerializableException.class)
+  public void serializationExceptionTest() throws ClassNotFoundException, IOException {
+    CacheKey cacheKey = new CacheKey();
+    cacheKey.update(new Object());
+    canSerialize(cacheKey);
+  }
+
+  @Test
+  public void serializationTest() throws ClassNotFoundException, IOException {
+    CacheKey cacheKey = new CacheKey();
+    cacheKey.update("serializable");
+    canSerialize(cacheKey);
+  }
+
+  private void canSerialize(final CacheKey object) throws ClassNotFoundException, IOException {
+      FileOutputStream fout = new FileOutputStream("target/address.ser");
```

Comment:
No need to use files - `ByteArrayOutputStream` and `ByteArrayInputStream` will do just fine.